pkg/installer/installer.go

```diff
@@ -89,227 +80,33 @@ type Installer struct {
 }
 
 // NewInstaller returns an Installer object with the kubectl client and in-memory filesystem
-func NewInstaller(config *apiv1.KubectlStorageOSConfig) (*Installer, error) {
+func NewInstaller(config *apiv1.KubectlStorageOSConfig, ensureNamespace bool) (*Installer, error) {
 	installer := &Installer{}
-	kubectlClient := otkkubectl.New()
-	fileSys, err := buildInstallerFileSys(config)
-	if err != nil {
-		return installer, err
-	}
+
 	clientConfig, err := pluginutils.NewClientConfig()
 	if err != nil {
 		return installer, err
 	}
 
-	installer = &Installer{
-		kubectlClient: kubectlClient,
-		clientConfig:  clientConfig,
-		fileSys:       fileSys,
-	}
-
-	return installer, nil
-}
-
-// buildInstallerFileSys builds an in-memory filesystem for installer with relevant storageos and
-// etcd manifests. If '--skip-etcd-install' flag is set, etcd dir is not created.
-// - storageos
-//   - operator
-//     - storageos-operator.yaml
-//     - kustomization.yaml
-//   - cluster
-//     - storageos-cluster.yaml
-//     - kustomization.yaml
-// - etcd
-//   - operator
-//     - etcd-operator.yaml
-//     - kustomization.yaml
-//   - cluster
-//     - etcd-cluster.yaml
-//     - kustomization.yaml
-func buildInstallerFileSys(config *apiv1.KubectlStorageOSConfig) (filesys.FileSystem, error) {
-	fs := filesys.MakeFsInMemory()
-	fsData := make(fsData)
-	stosSubDirs := make(map[string]map[string][]byte)
-
-	// build storageos/operator
-	stosOpFiles, err := createFileData(config.Spec.Install.StorageOSOperatorYaml, pluginversion.OperatorLatestSupportedURL(), stosOperatorFile)
-	if err != nil {
-		return fs, err
-	}
-	stosSubDirs[operatorDir] = stosOpFiles
-
-	// build storageos/cluster
-	stosClusterFiles, err := createFileData(config.Spec.Install.StorageOSClusterYaml, pluginversion.ClusterLatestSupportedURL(), stosClusterFile)
-	if err != nil {
-		return fs, err
-	}
-
-	// append storageos secret yaml to cluster yaml if necessary. This will happen in the event of an
-	// uninstall of storageos version < 2.5.0.
-	if config.InstallerMeta.StorageOSSecretYaml != "" {
-		stosSecretYaml, err := pullManifest(config.InstallerMeta.StorageOSSecretYaml)
+	if ensureNamespace {
```

Comment:
I'm not sure why this `ensureNamespace` functionality is necessary.

If the user passes a storageos operator namespace to the CLI, the namespace will be set in the operator's kustomization file here https://github.com/storageos/kubectl-storageos/blob/186f0b2ba372d4836cb7352fbbf4ec7a2c5d329b/pkg/installer/install.go#L121 and kustomized/applied here (creating the namespace if it does not already exist) https://github.com/storageos/kubectl-storageos/blob/186f0b2ba372d4836cb7352fbbf4ec7a2c5d329b/pkg/installer/install.go#L171
Am I missing something?

Reply:
Yes, but job needed to fetch manifest and namespace needed to spin up the job.

Reply:
Ah I see.. Chicken and egg! 🐔🥚😄